portalnet/src/overlay/service.rs

```diff
@@ -2497,6 +2496,14 @@ where
             return Some(entry.value().clone().enr());
         }
 
+        if let kbucket::Entry::Pending(mut entry, _) = self.kbuckets.write().entry(&key) {
```

Comment:
I'd add a comment explaining the purpose/motivation for checking kbuckets before `find_enr`